ops/testing.py

```diff
@@ -1546,7 +1546,14 @@ def make_dir(
             )
 
     def remove_path(self, path: str, *, recursive: bool = False):
-        file_or_dir = self._fs.get_path(path)
+        try:
+            file_or_dir = self._fs.get_path(path)
+        except FileNotFoundError:
+            if recursive:
+                return
+            raise pebble.PathError(
+                'generic-file-error', 'cannot remove non-existing path without recursive=True')
```

Comment:
As shown in my [latest comment on the issue](https://github.com/canonical/operator/issues/676#issuecomment-1009399323), the "kind" should be `not-found`. Pebble also formats the message differently, and I think it makes sense to match that (that that's not as important as kind) -- that would be `remove {path}: no such file or directory`.

Reply:
Updated per your comments, thanks!
I saw it mentioned in `./internal/daemon/api_files_test.go`.

Reply:
Are you referring to this in `TestRemoveMultiple`?

```
	checkFileResult(c, r.Result[2], tmpDir+"/non-empty", "generic-file-error", ".*directory not empty")
```

If so, that's a different error than `not-found`. There's no specific kind for "directory not empty", so it uses `generic-file-error` instead.